src/display/node_stream.js

```diff
@@ -288,8 +260,29 @@ class PDFNodeStreamRangeReader extends BaseRangeReader {
     this._request.end();
   }
 
+  read() {
+    return this._readCapability.promise.then(() => {
+      if (this._done) {
+        return Promise.resolve({ value: undefined, done: true, });
+      }
+      if (this._errored) {
+        return Promise.reject(this._reason);
+      }
+
+      let chunk = this._rangeRequest.read();
```

Comment:
> What stops you from putting read in the base class?

Sorry, but not sure why when I am doing something like https://github.com/mozilla/pdf.js/pull/8712#discussion_r131416557, I am unable to read from the stream, see https://github.com/mozilla/pdf.js/pull/8712#issuecomment-320933800.

Also `_read() { return this._rangeReques.read() }` creates the same issue.